packages/ckeditor5-dev-changelog/src/utils/getnewversion.ts

```diff
@@ -0,0 +1,30 @@
+/**
+ * @license Copyright (c) 2003-2025, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+import chalk from 'chalk';
+import type { ReleaseType } from 'semver';
+import type { SectionsWithEntries } from '../types.js';
+import { provideNewVersionForMonorepository } from './providenewversionformonorepository.js';
+import { logInfo } from './loginfo.js';
+
+/**
+ * Determines the next version number based on the changes and current version.
+ * This function analyzes the changes and suggests the appropriate version bump.
+ */
+export async function getNewVersion( sectionsWithEntries: SectionsWithEntries, oldVersion: string, packageName: string ): Promise<string> {
+	logInfo( `📍 ${ chalk.cyan( 'Determining the new version...' ) }\n` );
+
+	let bumpType: ReleaseType = 'patch';
+
+	if ( sectionsWithEntries.minor.entries.length || sectionsWithEntries.Feature.entries.length ) {
+		bumpType = 'minor';
+	}
+
+	if ( sectionsWithEntries.major.entries.length ) {
+		bumpType = 'major';
+	}
```

Comment:
This code expects that `sectionsWithEntries.minor` and `sectionsWithEntries.major` are always arrays, but this is not true. These properties may not exist.

Reply:
`sectionsWithEntries[ 'minor' | 'major' | 'feature' | 'fix' | 'other' | 'invalid' ]` will always exist. `entries` can be sometimes empty.

![image](https://github.com/user-attachments/assets/e8de2266-2539-4800-909c-36ceb582ec26)

Reply:
The change is added in [74596a5](https://github.com/ckeditor/ckeditor5-dev/pull/1090/commits/74596a55e43a4e5dcbc1caab02de84d24dfe174f).